src/routes/plans/[id]/+page.svelte

```diff
@@ -120,6 +125,9 @@
   }
 
   $: if ($plan) {
+    // Ensure there is no selected activity if the user came from another plan
+    selectActivity(null, null);
```

Comment:
Could this potentially cause an issue if we allow plan editing? I can see this being an issue if someone renames a plan or changes the time range while someone is viewing the same plan with an activity selected and having it just deselected randomly.

Reply:
Yeah that's a good point. Maybe we could get away with just keying off plan id instead of the whole plan object?

Reply:
Yeah agreed, I think plan id would be a good way